src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -360,6 +354,12 @@ public override uint EnableSsl(uint options)
         {
             _validateCert = (options & TdsEnums.SNI_SSL_VALIDATE_CERTIFICATE) != 0;
 
+            if (_sslOverTdsStream == null || _sslStream == null)
```

Comment:
Why did you move the ssl initialization to Connect() ? What happens to the ssl needs of the socket coming from ParallelConnect in case of MSF?